sdk/src/main/java/com/hedera/hashgraph/sdk/ContractCreateFlow.java

```diff
@@ -0,0 +1,342 @@
+     * Use this instead of {@link #setConstructorParameters(ContractFunctionParameters)} if you have already
+     * pre-encoded a solidity function call.
+     *
+     * @param constructorParameters The constructor parameters
+     * @return {@code this}
+     */
+    public ContractCreateFlow setConstructorParameters(byte[] constructorParameters) {
+        this.constructorParameters = constructorParameters;
+        return this;
+    }
+
+    /**
+     * Sets the parameters to pass to the constructor.
+     *
+     * @param constructorParameters The contructor parameters
+     * @return {@code this}
+     */
+    public ContractCreateFlow setConstructorParameters(ContractFunctionParameters constructorParameters) {
+        Objects.requireNonNull(constructorParameters);
+        return setConstructorParameters(constructorParameters.toBytes(null).toByteArray());
+    }
+
+    public String getContractMemo() {
+        return contractMemo;
+    }
+
+    /**
+     * Sets the memo to be associated with this contract.
+     *
+     * @param memo The String to be set as the memo
+     * @return {@code this}
+     */
+    public ContractCreateFlow setContractMemo(String memo) {
+        Objects.requireNonNull(memo);
+        contractMemo = memo;
+        return this;
+    }
+
+    @Nullable
+    public List<AccountId> getNodeAccountIds() {
+        return nodeAccountIds;
+    }
+
+    /**
+     * Set the account IDs of the nodes that this transaction will be submitted to.
+     * <p>
+     * Providing an explicit node account ID interferes with client-side load balancing of the
+     * network. By default, the SDK will pre-generate a transaction for 1/3 of the nodes on the
+     * network. If a node is down, busy, or otherwise reports a fatal error, the SDK will try again
+     * with a different node.
+     *
+     * @param nodeAccountIds The list of node AccountIds to be set
+     * @return {@code this}
+     */
+    public ContractCreateFlow setNodeAccountIds(List<AccountId> nodeAccountIds) {
+        Objects.requireNonNull(nodeAccountIds);
+        this.nodeAccountIds = nodeAccountIds;
+        return this;
+    }
+
+    private void splitBytecode() {
+        if(bytecode.length() > FILE_CREATE_MAX_BYTES) {
+            createBytecode = bytecode.substring(0, FILE_CREATE_MAX_BYTES);
+            appendBytecode = bytecode.substring(FILE_CREATE_MAX_BYTES);
+        } else {
+            createBytecode = bytecode;
+            appendBytecode = "";
+        }
+    }
+
+    private FileCreateTransaction spawnFileCreateTransaction(Client client) {
+        var fileCreateTx = new FileCreateTransaction()
+            .setKeys(Objects.requireNonNull(client.getOperatorPublicKey()))
+            .setContents(createBytecode);
+        if (nodeAccountIds != null) {
+            fileCreateTx.setNodeAccountIds(nodeAccountIds);
+        }
+        return fileCreateTx;
+    }
+
+    private FileAppendTransaction spawnFileAppendTransaction(FileId fileId) {
```

Comment:
I think spawn is the wrong terminology as in technology that's usually used to indicate a process has been forked and started. Just `fileAppendTransaction()` or `getFileAppendTransaction()` would fit better. Same for other spawn methods.

Reply:
How about `make*()`?

Reply:
Or `create*()`?

Reply:
That works.